zebra-state/src/service/finalized_state/disk_format/block.rs

```diff
@@ -90,17 +113,22 @@ impl FromDisk for Block {
 }
 
 impl IntoDisk for Height {
-    type Bytes = [u8; 4];
+    /// Consensus: see the note at [`MAX_ON_DISK_HEIGHT`].
+    type Bytes = [u8; 3];
 
     fn as_bytes(&self) -> Self::Bytes {
-        self.0.to_be_bytes()
+        let mem_bytes = self.0.to_be_bytes();
+
+        let disk_bytes = truncate_zero_be_bytes(&mem_bytes, Height::fixed_disk_byte_len());
+
+        disk_bytes.try_into().unwrap()
     }
 }
 
 impl FromDisk for Height {
     fn from_bytes(bytes: impl AsRef<[u8]>) -> Self {
-        let array = bytes.as_ref().try_into().unwrap();
-        Height(u32::from_be_bytes(array))
+        let mem_bytes = expand_zero_be_bytes(bytes.as_ref(), 4);
+        Height(u32::from_be_bytes(mem_bytes.try_into().unwrap()))
     }
 }
```

Comment:
Add constants for disk byte size and memory byte size, to make the difference explicit.

Reply:
Fixed in 2b84de1dd